src/machine/machine_stm32f407.go

```diff
@@ -7,44 +7,31 @@ package machine
 import (
 	"device/stm32"
 	"runtime/interrupt"
+	"unsafe"
 )
 
 func CPUFrequency() uint32 {
 	return 168000000
 }
 
 const (
-	// Mode Flag
-	PinOutput        PinMode = 0
-	PinInput         PinMode = PinInputFloating
-	PinInputFloating PinMode = 1
-	PinInputPulldown PinMode = 2
-	PinInputPullup   PinMode = 3
-
-	// for UART
-	PinModeUartTX PinMode = 4
-	PinModeUartRX PinMode = 5
-
-	//GPIOx_MODER
-	GPIO_MODE_INPUT          = 0
-	GPIO_MODE_GENERAL_OUTPUT = 1
-	GPIO_MODE_ALTERNABTIVE   = 2
-	GPIO_MODE_ANALOG         = 3
-
-	//GPIOx_OTYPER
-	GPIO_OUTPUT_MODE_PUSH_PULL  = 0
-	GPIO_OUTPUT_MODE_OPEN_DRAIN = 1
-
-	// GPIOx_OSPEEDR
-	GPIO_SPEED_LOW     = 0
-	GPIO_SPEED_MID     = 1
-	GPIO_SPEED_HI      = 2
-	GPIO_SPEED_VERY_HI = 3
-
-	// GPIOx_PUPDR
-	GPIO_FLOATING  = 0
-	GPIO_PULL_UP   = 1
-	GPIO_PULL_DOWN = 2
+	// Alternative peripheral pin functions
+	AF0_SYSTEM                AltFunc = 0
+	AF1_TIM1_2                        = 1
+	AF2_TIM3_4_5                      = 2
+	AF3_TIM8_9_10_11                  = 3
+	AF4_I2C1_2_3                      = 4
+	AF5_SPI1_SPI2                     = 5
+	AF6_SPI3                          = 6
+	AF7_USART1_2_3                    = 7
+	AF8_USART4_5_6                    = 8
+	AF9_CAN1_CAN2_TIM12_13_14         = 9
+	AF10_OTG_FS_OTG_HS                = 10
+	AF11_ETH                          = 11
+	AF12_FSMC_SDIO_OTG_HS_1           = 12
+	AF13_DCMI                         = 13
+	AF14                              = 14
+	AF15_EVENTOUT                     = 15
```

Comment:
Same here as for `GPIOModeInput`, it seems better to me to move these constants over to src/device/stm32.